src/core/metallicroughnesseffect.cpp

```diff
@@ -718,8 +719,25 @@ void MetallicRoughnessEffect::setToneMappingAlgorithm(MetallicRoughnessEffect::T
     emit toneMappingAlgorithmChanged(algorithm);
 }
 
+Qt3DRender::QAbstractTexture *MetallicRoughnessEffect::brdfLUT() const
+{
+    return m_brdfLUTParameter->value().value<Qt3DRender::QAbstractTexture *>();
+}
+
 void MetallicRoughnessEffect::setBrdfLUT(Qt3DRender::QAbstractTexture *brdfLUT)
 {
+    auto d = Qt3DCore::QNodePrivate::get(this);
+    auto current = this->brdfLUT();
+    if (current) {
+        d->unregisterDestructionHelper(current);
+        if (current->parentNode() == this)
```

Comment:
you don't need to unparent, the old one (who manages its lifetime then ?) You should parent the new brdfLUT to make sure the  Qt3D backend creates it (which we do by setting the parent).

Reply:
ah, yes. But should I delete if the parent is this?